pkg/apis/messaging/v1alpha1/subscription_types.go

```diff
@@ -114,11 +114,17 @@ type SubscriptionSpec struct {
 // ReplyStrategy specifies the handling of the Subscriber's returned replies.
 // If no Subscriber is specified, the identity function is assumed.
 type ReplyStrategy struct {
+	//  The resource pointed by this Destination must meet the Addressable contract
+	//  with a reference to the Addressable duck type. If the resource does not meet this contract,
+	//  it will be reflected in the Subscription's status.
+	// +optional
+	Destination *apisv1alpha1.Destination `json:",inline"`
```

Comment:
I think what is happening here is go promotes the type as the field name automatically, but I think we should make it clear that we do not intend to have a named field in the future. 

If you change this to:

```go
Destination *apisv1alpha1.Destination `json:",inline"`
```

Then we can do things like `subscription.Spec.Reply.URI` and this will just work (it will also work with `subscription.Spec.Reply.Destination.URI` because of go). 

At the moment it only will work with  `subscription.Spec.Reply.Destination.URI`

Reply:
fixed.